components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -2068,6 +2068,25 @@ public static String getTenantDomainOfOauthApp(String clientId)
         return getTenantDomainOfOauthApp(oAuthAppDO);
     }
 
+    /**
+     * Get the client secret of the application.
+     *
+     * @param consumerKey Consumer Key provided by the user.
+     * @return Consumer Secret.
+     * @throws IdentityOAuth2Exception Error when loading the application.
+     * @throws InvalidOAuthClientException Error when loading the application.
+     */
+    public static String getClientSecret(String consumerKey) throws IdentityOAuth2Exception,
+            InvalidOAuthClientException {
+
+        OAuthAppDO oAuthAppDO;
+        oAuthAppDO = OAuth2Util.getAppInformationByClientId(consumerKey);
+        if (oAuthAppDO == null) {
```

Comment:
Can we end up with null?

Reply:
In the DAO, if the consumerKey is not found in the DB, we may get null. In the token flows, we are validating the consumerKey in the earlier stages. But as this is Utils, I have added this check.